chromium_src/chrome/install_static/chromium_install_modes.cc

```diff
@@ -137,6 +146,38 @@ const InstallConstants kInstallModes[] = {
         IDR_SXS,                               // App icon resource id.
     },
 };
+#else
+const InstallConstants kInstallModes[] = {
+    // The primary (and only) install mode for Brave developer build.
+    {
+        sizeof(kInstallModes[0]),
+        DEVELOPER_INDEX,  // The one and only mode for developer mode.
+        "",              // No install switch for the primary install mode.
+        L"",             // Empty install_suffix for the primary install mode.
+        L"",             // No logo suffix for the primary install mode.
+        L"",          // Empty app_guid since no integraion with Google Update.
+        L"Brave Developer",  // A distinct base_app_name.
+        L"BraveDeveloper",  // A distinct base_app_id.
+        L"BraveDelveloperHTM",                             // ProgID prefix.
+        L"Brave Developer HTML Document",                  // ProgID description.
+        L"{7D2B3E1D-D096-4594-9D8F-A6667F12E0AC}",  // Active Setup GUID.
```

Comment:
Just copied guids from upstream. which value should I use?

Reply:
You should use the ones above as I created all new IDs for them.

Reply:
And made sure other places that used those IDs were also updated.

Reply:
I think when we compile with no brave chromium override we could skip the cc_wrapper check for override files.   That way we don't need to maintain separate chrome stuff in our src/brave dir.

Reply:
Oops..  I didn't know about our cc_wrapper.
That script replaces original file with our files when same file exists.
So, My thought was wrong.
`install_static_util` target includes our version and other targets  that depends on `//brave:browser_dependencies` also includes ours , too.

I want to make only `install_static_util` target dependent on our `chromium_install_modes files.*'.
This cl's approach does that, but, upstream BUILD.gn file needs to be modified.

Hmm. I need to think a little bit more how to achieve this with cc_wrapper.

Regarding to guid, should I use same ID above you created in official build gard? or need to generate new one?

Reply:
When we use our `redirect-cc.py`, there is one problem - dependency check.
To trigger re-build, original file is touched if we don't add additional dependency.
In `install_static_util` case, we added another dependency to `brave/browser_dependencies`.
How about making dependency explicitly instead of using cc-wrapper like below?
```
else {
  if (brave_chromium_build) {
    sources += [
      "//brave/chromium_src/chrome/install_static/chromium_install_modes.cc",
      "//brave/chromium_src/chrome/install_static/chromium_install_modes.h",
    ]
  } else {
    sources += [
      "chromium_install_modes.cc",
      "chromium_install_modes.h",
    ]
  }
```
With this, I think we don't need to introduce another dependency and can know dependency easily.

Reply:
> Regarding to guid, should I use one of IDs above you created in official build guard? or need to generate new one?

No they were already all newly generated, no need to generate any new IDs here.

Reply:
Good point about the dependencies.  The whole reason to introduce cc_wrapper hack was to avoid having to patch extra gn files with the real source path though.  Maybe we can just have the dev changing these files touch the original ones too, the overrides should very rarely be used.

@bridiver thoughts?

Reply:
@bbondy , updated. Please check again.

Reply:
I think in an ideal world we would have a way to reference gn targets and add/remove source files without actually modifying them. For the cc_wrapper files we can create a dummy script target and mark them as inputs to a no-op script or something like that